lib/cn/omf.h

```diff
@@ -226,21 +229,35 @@ OMF_SETGET(struct wbt_lfe_omf, lfe_kmd, 16)
 
 /*****************************************************************
  *
- * Vblock header OMF
+ * Vblock footer OMF
  *
  ****************************************************************/
 
-#define VBLOCK_HDR_MAGIC ((u32)0xea73feed)
+#define VBLOCK_FOOTER_MAGIC ((u32)0xea73feed)
+#define VBLOCK_FOOTER_LEN   4096
 
-/* Version 2 header */
-struct vblock_hdr_omf {
-    uint32_t vbh_magic;
-    uint32_t vbh_version;
-    uint64_t vbh_vgroup;
+/*
+ * min_key is stored at offset VBLOCK_FOOTER_LEN - (2 * HSE_KVS_KEY_LEN_MAX)
+ * min key is inclusive in all the vblocks
+ *
+ * max_key is stored at offset VBLOCK_FOOTER_LEN - HSE_KVS_KEY_LEN_MAX
+ * max key is exclusive in all but the last vblock
+ */
+struct vblock_footer_omf {
+    uint32_t vbf_magic;
+    uint32_t vbf_version;
+    uint64_t vbf_vgroup;
+    uint32_t vbf_min_klen;
+    uint32_t vbf_max_klen;
```

Comment:
Presumably the remainder of the block which contains the footer is zeroed out so that we can append new fields down the road and be certain they are initialized to zero?

Reply:
Good point, fixed it now. Thanks!